Builder/Generator.php

```diff
@@ -118,44 +118,50 @@ public function addBuilder(BuilderInterface $builder)
     protected function mergeParameters(array $global, array $builder)
     {
         foreach ($global as $param => &$value) {
-            if (array_key_exists($param, $builder)) {
-                if (in_array($param, array('fields', 'actions', 'object_actions', 'batch_actions'))) {
-                    // Grab builder configuration only if defined
-                    if (!is_null($builder[$param])) {
-                        $configurations = array();
-                        foreach ($builder[$param] as $name => $configuration) {
-                            if (is_array($configuration) || is_null($configuration)) {
-                                if (!is_null($value) && array_key_exists($name, $value)) {
-                                    $configurations[$name] = $configuration
-                                        ? $this->mergeConfiguration($value[$name], $configuration) // Override definition
-                                        : $value[$name]; // Configuration is null => use global definition
-                                } else {
-                                    // New definition (new field, new action) from builder
-                                    $configurations[$name] = $configuration;
-                                }
-                            } else {
-                                throw new \InvalidArgumentException(
-                                    sprintf('Invalid %s "%s" builder definition for %s', $param, $name, $this->getFromYaml('params.model'))
-                                );
-                            }
-                        }
-
-                        if (in_array($param, array('actions', 'object_actions', 'batch_actions'))) {
-                            // Actions list comes from builder
-                            $value = $configurations;
-                        } else {
-                            // All fields are still available in a builder
-                            $value = array_merge($value ?:array(), $configurations);
-                        }
-                    }
+            if (!array_key_exists($param, $builder)) {
+                continue;
+            }
+
+            if (!in_array($param, array('fields', 'actions', 'object_actions', 'batch_actions'))) {
+                if (is_array($value)) {
+                    $value = $this->recursiveReplace($value, $builder[$param]);
+                } else {
+                    $value = $builder[$param];
+                }
+
+                continue;
+            }
+
+            // Grab builder configuration only if defined
+            if (is_null($builder[$param])) {
+                continue;
+            }
+
+            $configurations = array();
+            foreach ($builder[$param] as $name => $configuration) {
+                if (!(is_array($configuration) || is_null($configuration))) {
```

Comment:
Maybe `if (!is_array($configuration) && !is_null($configuration)) {` would be more clear whats going on here